### PR TITLE
feat: expose system preferences to utility process

### PR DESCRIPTION
**Pull request by georgexu99**
#### Description of Change

This PR exposes `systemPreferences` to the utility process, rather than only in the main process.

#### Motive
`systemPreferences.getMediaAccessStatus` only returns accurate permissions-states at the start of a user's session, [but may not reflect permission changes made to the app until the app full quits/restarts.](https://github.com/electron/electron/issues/36722#issuecomment-1362740564) This can lead to silent failures when we attempt to access a device when the permissions have been revoked mid-session. This fixes this problem on macOS

Being able to call this function from within the Utility Module will allow for accurate permission state retrieval.
#### Testing
[x] added test
[x] manually verified change on macOS with [gist](https://gist.github.com/37fe7624d27506da0852034f51de56f1)

#### Release Notes

Notes: expose `systemPreferences` to `utilityProcess`

### CODE REVIEW

File: lib/utility/api/module-list.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // Utility side modules, please sort alphabetically.
 export const utilityNodeModuleList: ElectronInternal.ModuleEntry[] = [
-  { name: 'net', loader: () => require('./net') }
+  { name: 'net', loader: () => require('./net') },
+  { name: 'systemPreferences', loader: () => require('./system-preferences') }
```
Comment (MarshallOfSound) [inline lib/utility/api/module-list.ts:4]: Can we just map this to the `@browser` file? no need to copy paste the impl I think

Comment (georgexu99) [reply lib/utility/api/module-list.ts]: done- uses @electron/internal/browser/api/system-preferences now. However, the net module and some others all re-copy: any insights as to why we chose to do that

**Review by erickzhao — APPROVED** [review]
API LGTM

**Review by VerteDinde — APPROVED** [review]
Looks great!

**Review by jkleinsc — APPROVED** [review]
API LGTM

**Pull request merged**